data/campaigns/Eastern_Invasion/scenarios/04c_Mal-Ravanals_Capital.cfg

```diff
@@ -4,145 +4,421 @@
+                [/message]
+                [message]
+                    speaker={ID}
+                    message= _ "Five of my comrades are still imprisoned. We will gladly enter your service in appreciation for our freedom."
+                [/message]
+            [/then]
+        [/if]
+        {VARIABLE first_prisoner_freed yes}
+        [if]
+            [not]
+                [have_unit]
+                    side=8
+                [/have_unit]
+            [/not]
 
-    {LIMIT_CONTEMPORANEOUS_RECRUITS 5 Shadow 1}
-    {LIMIT_CONTEMPORANEOUS_RECRUITS 5 Deathblade 3}
+            [then]
+                [set_achievement]
+                    content_for=eastern_invasion
+                    id=ei_S04c
+                [/set_achievement]
+            [/then]
+        [/if]
+    [/event]
+#enddef
+    [event] # this is a really ugly way to have an additional line for Terraent
+        name=rescued_Terraent # fired with rescued_{ID}
+        [message]
+            speaker=Gweddry
+            message=_ "A sun paladin! I had no idea there were any of your order still in Wesnoth."
+        [/message]
+    [/event]
+    [event]
+        name=prestart
+        {NAMED_NOTRAIT_UNIT 8 (Knight) 22 3 Aeraery (_"Aeraery")}
+        [+unit]
+            [modifications]
+                {TRAIT_SURVIVOR}
+                {TRAIT_QUICK}
+            [/modifications]
+        [/unit]
+        {IMPRISON Aeraery     21 3 (Thank you! My poor horse could barely breathe trapped among these reeds!)}
+
+        {NAMED_NOTRAIT_UNIT 8 (Knight) 34 2 Teoryn (_"Teoryn")}
+        [+unit]
+            [modifications]
+                {TRAIT_SURVIVOR}
+                {TRAIT_STRONG}
+            [/modifications]
+        [/unit]
+        {IMPRISON Teoryn      33 2 (Bah, I didn’t need your help! Sooner or later, I would have gotten out of this cage on my own.)}
+
+        {NAMED_NOTRAIT_UNIT 8 (Knight) 7 15 Yrthynaenc (_"Yrthynaenc")}
+        [+unit]
+            [modifications]
+                {TRAIT_SURVIVOR}
+                {TRAIT_QUICK}
+            [/modifications]
+        [/unit]
+
+        {IMPRISON Yrthynaenc  7 16 (My thanks, friends. I had despaired of ever again riding free amongst the living.)}
+        {NAMED_NOTRAIT_UNIT 8 (Knight) 24 21 Inyc (_"Inyc")}
+        [+unit]
+            [modifications]
+                {TRAIT_SURVIVOR}
+                {TRAIT_RESILIENT}
+            [/modifications]
+        [/unit]
+        {IMPRISON Inyc        25 22 (Vengence upon the undead! I will make them pay for the abuse they’ve inflicted upon us!)}
+
+        {NAMED_NOTRAIT_UNIT 8 (Knight) 42 21 Vobreddaent (_"Vobreddaent")}
+        [+unit]
+            [modifications]
+                {TRAIT_SURVIVOR}
+                {TRAIT_STRONG}
+            [/modifications]
+        [/unit]
+        {IMPRISON Vobreddaent 41 21 (Truly you are heroes of legend to have fought so deep past the undead! I am forever in your debt.)}
```

Comment:
All of the IMPRISON macros are broken. These lines need to be marked translatable.

Reply:
Yep, steve and I noted this on line 354.